addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/httputils/HtmlContextAnalyser.java

```diff
@@ -135,6 +137,123 @@ public List<HtmlContext> getHtmlContexts(
+     * hashmap We follow a left to right approach for parsing the tags. We start at the beginning of
+     * the string and look for the first '<' We then look for the first '>' Then we remove the
+     * parsed portion from the string and we reiterate.
+     */
+    public static HashMap<String, HashMap<String, String>> parseTag(String input) {
+        HashMap<String, HashMap<String, String>> tagMap =
+                new HashMap<String, HashMap<String, String>>();
+        HashMap<String, String> attMap = new HashMap<String, String>();
+        int tagStart = 0;
+        int tagEnd = 0;
+        int firstSpace = 0;
+        String attr = null;
+        String tagString = null;
+        String tagName = null;
+
+        while (input.length() > 0) {
+            tagStart = input.indexOf('<');
+            if (tagStart == -1) {
+                break;
+            }
+            tagEnd = input.indexOf('>');
+            if (tagEnd == -1) {
+                break;
+            }
+            tagString = StringUtils.strip(input.substring(tagStart + 1, tagEnd));
+            /** Check if tag string is comment or not. If it is a comment then don't parse it. */
+            if (tagString.startsWith("!--")) {
+                input = input.substring(tagEnd + 1);
+                continue;
+            } else if (tagString.charAt(0) != '/') {
+                firstSpace = tagString.indexOf(' ');
+                if (firstSpace != -1) {
+                    tagName =
+                            StringUtils.strip(
+                                    input.substring(tagStart + 1, tagStart + firstSpace + 1));
+                    attr = StringUtils.strip(input.substring(tagStart + firstSpace + 1, tagEnd));
+                    attMap = parseAttributes(attr);
+                } else {
+                    tagName = tagString;
+                }
+            } else {
+                tagName = StringUtils.strip(input.substring(tagStart + 1, tagEnd));
+            }
+            if (tagName != null && tagName.charAt(0) != '/') {
+                tagMap.put(tagName, attMap);
+            }
+            input = input.substring(tagEnd + 1);
+        }
+        return tagMap;
+    }
+
+    public static HashMap<String, String> parseAttributes(String attr) {
+        HashMap<String, String> attMap = new HashMap<String, String>();
+        String content = attr;
+        /**
+         * there can be multiple attributes in the input string like: onclick="alert('hi')"
+         * accesskey="x" b="" so we need to loop through the string and find the attributes we then
+         * parse the attributes and add them to the hashmap We follow a left to right approach for
+         * parsing the attributes. We start at the beginning of the string and look for the first
+         * '=' We then look for the first '"' Then we remove the parsed portion from the string and
+         * we reiterate.
+         */
```

Comment:
I think of dropping these comments. Like I added it for me so that I could think out loud and also provide you guys with an insight into the logic. :)

Reply:
Hmmm I have fixed feelings. I like being able to see the previous person’s line of thought when I have to maintain something … but maybe it also isn’t necessary. Let’s let someone else on the team chime in 😉

Reply:
I thought the same. A part of me feels it's unnecessary :). A part of me wants to keep it if someday something bad happens - This logic can be improved/fixed easily :) - even though I'm writing tests to cover all of the edge cases possible and this shouldn't / wouldn't break :).

Reply:
I didn't check the code yet so the reason might be obvious, why aren't we leveraging Jericho to get the attributes?

Reply:
I did try to parse the inner HTML or custom-named tags eg "tag". But I couldn't parse it. As for the HTML body if I did getChildElements() I got nothing :)

Reply:
When I do element.getContent().toString()  .. the string contains "<body> <{OW4SPZ4P in 1337}> </body>" where the tag has the eyecatcher name .... so the eyecatcher is present in the element but not parsed as a child

Reply:
Could you provide concrete examples? Add tests that show what can be parsed with this method that can't with Jericho.

Reply:
How can I show that? Like literally the firing range test HTML that I've added cannot be parsed using Jericho when the eye catcher is inserted  :). 
I need some help to show this "test". Can be a useful for others too :)

Reply:
like what should I be naming the test and carrying out in it ? :)

Reply:
Just call the method with that HTML, the outcome is that it returns the expected tags/attributes. For the name, `shouldExtractAttributesFromMalformedHtml` (if that's the case).

Reply:
The tests can replace the code comments btw, there's nothing like seeing a test failing when you tinker with existing code.

Reply:
added a test in context analyser. PS ignore the comment lines I'll remove them with the Javadoc update and squash the commits :)